pkg/file/lazy_bounded_read_closer.go

```diff
@@ -72,3 +64,42 @@ func (d *lazyBoundedReadCloser) Close() error {
 	d.reader = nil
 	return err
 }
+
+func (d *lazyBoundedReadCloser) Seek(offset int64, whence int) (int64, error) {
+	if err := d.isFileOpen(); err != nil {
+		return 0, err
+	}
+
+	return d.reader.Seek(offset, whence)
+}
+
+func (d *lazyBoundedReadCloser) ReadAt(b []byte, off int64) (n int, err error) {
+	if err := d.isFileOpen(); err != nil {
```

Comment:
nit: since `isFileOpen` doesn't return a boolean and has side effects a better name might be `openFile`